dask/array/tests/test_rechunk.py

```diff
@@ -393,3 +393,164 @@ def test_dont_concatenate_single_chunks(shape, chunks):
     assert not any(funcname(task[0]).startswith('concat')
                    for task in dsk.values()
                    if dask.istask(task))
+
+
+def test_intersect_nan():
+    old_chunks = ((float('nan'), float('nan')), (8,))
+    new_chunks = ((float('nan'), float('nan')), (4, 4))
+
+    result = list(intersect_chunks(old_chunks, new_chunks))
+    expected = [(((0, slice(0, None, None)), (0, slice(0, 4, None))),
+                ((1, slice(0, None, None)), (0, slice(0, 4, None)))),
+                (((0, slice(0, None, None)), (0, slice(4, 8, None))),
+                ((1, slice(0, None, None)), (0, slice(4, 8, None)))),
+                (((1, slice(0, None, None)), (0, slice(0, 4, None))),),
+                (((1, slice(0, None, None)), (0, slice(4, 8, None))),)]
+    assert result == expected
+
+
+def test_intersect_nan_single():
+    old_chunks = ((float('nan'),), (10,))
+    new_chunks = ((float('nan'),), (5, 5))
+
+    result = list(intersect_chunks(old_chunks, new_chunks))
+    expected = [(((0, slice(0, None, None)), (0, slice(0, 5, None))),),
+                (((0, slice(0, None, None)), (0, slice(5, 10, None))),)]
+    assert result == expected
+
+
+def test_intersect_nan_long():
+
+    old_chunks = (tuple([float('nan')] * 4), (10,))
+    new_chunks = (tuple([float('nan')] * 4), (5, 5))
+    result = list(intersect_chunks(old_chunks, new_chunks))
+    expected = [
+        (((0, slice(0, None, None)), (0, slice(0, 5, None))),
+         ((1, slice(0, None, None)), (0, slice(0, 5, None)))),
+        (((0, slice(0, None, None)), (0, slice(5, 10, None))),
+         ((1, slice(0, None, None)), (0, slice(5, 10, None)))),
+        (((1, slice(0, None, None)), (0, slice(0, 5, None))),),
+        (((1, slice(0, None, None)), (0, slice(5, 10, None))),),
+        (((1, slice(0, None, None)), (0, slice(0, 5, None))),),
+        (((1, slice(0, None, None)), (0, slice(5, 10, None))),),
+        (((1, slice(0, None, None)), (0, slice(0, 5, None))),),
+        (((1, slice(0, None, None)), (0, slice(5, 10, None))),)]
+    assert result == expected
+
+
+def test_rechunk_unknown_from_pandas():
+    dd = pytest.importorskip('dask.dataframe')
+    pd = pytest.importorskip('pandas')
+
+    x = dd.from_pandas(pd.DataFrame(np.random.randn(50, 10)), 2).values
+    # result = x.rechunk({1: 5})
+    result = x.rechunk((None, (5, 5)))
+    assert np.isnan(x.chunks[0]).all()
+    assert np.isnan(result.chunks[0]).all()
+    assert result.chunks[1] == (5, 5)
+
+
+def test_rechunk_unknown_from_array():
+    import dask; dask.set_options(get=dask.get)  # noqa
```

Comment:
This is causing test failures.  You probably want this to be under a context manager.  Or it looks like you might not need this at all.  Typically we use `assert_eq` when we need to compute things, this automatically uses the synchronous scheduler (and checks many other useful things).